app/assets/javascripts/views/file_uploader_view.js.coffee

```diff
@@ -38,30 +44,39 @@ class ELMO.Views.FileUploaderView extends ELMO.Views.ApplicationView
     if confirm($(event.currentTarget).data('confirm-msg'))
       @$('.existing').remove()
       @$('.dropzone').show()
-      @idField.val('')
+      @clearMetaFields()
 
   fileUploaded: (responseData) ->
-    @idField.val(responseData.id)
+    for k, v of responseData
+      @$("input:hidden[name*=#{k}]").val(v)
 
   uploadErrored: (file, responseData) ->
     @dropzone.removeFile(file)
     errors = if responseData.errors
       responseData.errors.join("<br/>")
     else
-      I18n.t('errors.file_upload')
-    @$('.error-msg').show().html(errors)
+      I18n.t('errors.file_upload.error')
+    @$('.dz-error-msg').show().html(errors)
 
   fileRemoved: ->
     @$('.dz-message').show()
-    @idField.val('')
+    @clearMetaFields()
 
   uploadStarting: ->
     if @listener?
       @listener.uploadStarting()
-    @$('img')[0].src = @genericThumbPath
+    if @genericThumbPath?
+      @$('img')[0].src = @genericThumbPath
+    else
+      @$('img').hide()
+      @$('.dz-details').css('margin', '0px')
```

Comment:
is there a cleaner alternative here besides having two separate dropzone_preview templates?

Reply:
Yes. You can just add a class to one of the outer elements like "no-thumb" and then in the appropriate CSS file hide the image and remove the margin. Since the HTML is different in each case you can add it literally in the template instead of adding from JS using `addClass` or passing it as an option to a partial or whatever.